### PR TITLE
refactor kafka producer from being called a channel to a receiver.

### PR DESCRIPTION
**Pull request by travis-minke-sap**
Recently, in the distributed kafka channel implementation, we renamed the Kafka Producer pod's top level directory from `channel/` to `receiver` as the term "channel" was being abused/overused.  This PR finishes that renaming effort by changing the code to read consistently, and altering the runtime artifacts naming to include `receiver` instead of `channel`.  Other than naming, and some other general code clean-up, there are no intended behavioural differences.

### CODE REVIEW

**Comment by travis-minke-sap** [comment]
/hold

found additional touch-points

File: pkg/channel/distributed/controller/kafkasecret/reconciler.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, secret *corev1.Secret) re
 	r.logger.Debug("<==========  START KAFKA-SECRET FINALIZATION  ==========>")
 	logger := r.logger.With(zap.String("Secret", secret.Name))
 
-	// Reconcile The Affected KafkaChannel Status To Indicate The Channel Service/Deployment Are Not Longer Available
+	// Reconcile The Affected KafkaChannel Status To Indicate The Receiver Service/Deployment Are No Longer Available
```
Comment (eric-sap) [inline pkg/channel/distributed/controller/kafkasecret/reconciler.go:66]: As long as we're fixing grammar
```suggestion
	// Reconcile The Affected KafkaChannel Status To Indicate The Receiver Service/Deployment Is No Longer Available
```

File: pkg/channel/distributed/controller/util/channel.go [hunk 2]
```diff
@@ -44,19 +44,19 @@ func NewChannelOwnerReference(channel *kafkav1beta1.KafkaChannel) metav1.OwnerRe
...
 //
-func ChannelDnsSafeName(kafkaSecretName string) string {
+func ReceiverDnsSafeName(kafkaSecretName string) string {
 
 	// In order for the resulting name to be a valid DNS component it's length must be no more than 63 characters.
-	// We are consuming 17 chars for the component separators, hash, and Channel suffix, which reduces the
-	// available length to 46. We will allocate 41 characters to the kafka secret name leaving an extra buffer.
-	safeSecretName := GenerateValidDnsName(kafkaSecretName, 41, true, false)
+	// We are consuming 18 chars for the component separators, hash, and Receiver suffix, which reduces the
+	// available length to 46. We will allocate 40 characters to the kafka secret name leaving an extra buffer.
```
Comment (eric-sap) [inline pkg/channel/distributed/controller/util/channel.go:56]: math!
```suggestion
	// available length to 45. We will allocate 40 characters to the kafka secret name leaving an extra buffer.
```

**Review by eric-sap — COMMENTED** [review]
minor nits

File: pkg/channel/distributed/controller/util/channel.go (hunk 2 above)
Comment (travis-minke-sap) [reply pkg/channel/distributed/controller/util/channel.go]: 👍

File: pkg/channel/distributed/controller/kafkasecret/reconciler.go (hunk 1 above)
Comment (travis-minke-sap) [reply pkg/channel/distributed/controller/kafkasecret/reconciler.go]: well.... i guess....

**Review by eric-sap — APPROVED** [review]
looks good

**Pull request merged**